app/Http/Controllers/DashboardController.php

```diff
@@ -28,19 +28,17 @@ public function index()
             )->with('debts.contact')
             ->first();
 
-        $numberOfContacts = $account->contacts()
-                                    ->real()
-                                    ->active()
-                                    ->count();
+        $numberOfContacts = $account->addressBookContacts()
```

Comment:
Here I don't like it 😀
Why do we have to think about querying this method instead of `contacts`?

I think address books should be transparent unless we really want to manage it in the code/controller because we need to manage different address books.

In this case, the right call should be `account->contacts()` and that's it. We should manage the notion of address books in the `contacts()` method, and that way everything would be transparent.

Or perhaps it should be something like 

```
$account->contacts()
     ->real()
     ->active()
     ->count();
```

and we could have another scope to specify the address book when we need to scope this to a specific address book?

Reply:
That was my first try.. and it does not work like this. contacts() return a HasMany, and I can add a filter in this function. I don't know where to insert the scope/Query without explicitly call it. If you have a better idea ok, but I have zero.

Reply:
There is not "less complex code" where we code magically guess the address book currently seeing.

I guess I could just revert the changes and go back to something like
```
$account->contacts()
     ->addressbook($addressBookName)
     ->real()
     ->active()
     ->count();
```
If $addressBookName is null it will scope to default address book (null id), if not it will scope to the address book.